src/components/profile/EnvironmentForm.jsx

```diff
@@ -47,6 +49,7 @@ class EnvironmentForm extends Component {
       tag: '',
       tags: props.envExists && Boolean(props.environment.tags) ? props.environment.tags : [],
       error: '',
+      topic: props.envExists && Boolean(props.environment.topic) ? props.environment.topic.id : '',
```

Comment:
Just to clean this up a little I might deconstruct `props` so you don't have to repeatedly access it:
```
const { envExists, environment} = props;
this.state = {
  ...
}
```